### DISCUSSION TITLE
@formily/vue 通过mapProps和connect给Upload组件传入默认onChange参数时会有警告

### DISCUSSION THREAD

**Discussion by hi-hugo-dev** [alibaba/formily#1537] [Q&A 2.x]
**环境**
@formily/vue 2.0.0-beta.58
vue    3.0.10
and-design-vue    2.1.3

**描述**
@formily/vue 通过mapProps和connect传入Upload组件的props参数onChange时会有警告
而且实际上onChange方法也不生效，无法通过这个方法调整Upload组件上传后的fileList参数

**警告内容：**
index.js:27 [Vue warn]: Invalid prop: type check failed for prop "onChange". Expected Function, got Array 

**codesandbox链接：**
https://codesandbox.io/s/pensive-bartik-3qzz7?file=/src/App.vue

**问题定位**
在@formily/vue 的 Fiedl、RecursionField和ReactiveField中props参数流转都没问题，onChange参数一直是 function 原封不动。
但是传进vue的runtime-core后，onChange参数被转换为了一个数组（长度为2），数组第一个元素是 我定义的传入的onChange对象，第二个是 在ReactiveField组件中定义的change事件（event.change）。
![image](https://user-images.githubusercontent.com/40160972/120442020-c4049c00-c3b7-11eb-8961-b403df6b4e0b.png)
![image](https://user-images.githubusercontent.com/40160972/120443420-31fd9300-c3b9-11eb-9acd-5a51fce9ef22.png)
![image](https://user-images.githubusercontent.com/40160972/120443569-5c4f5080-c3b9-11eb-9860-22b257315962.png)

分析后感觉应该是 ReactiveField中传入了一个默认的onchange参数，是在生成vnode实例时就挂上的（这时候已经挂到vnode实例的props里了），而外部通过mapProps传入的是挂在 单独额外的props参数中的，通过render createVNode进行渲染，然后vue这块进行处理时将两个对象里的props合并变成一个数组了。

**结论**
目前这块不确定是bug还是写法问题，请教下要怎么处理

**Comment by hi-hugo-dev** [comment]
再补充下，组件渲染过程中，connect似乎会被重复多次调用，不确定是机制还是
export const Upload: any = connect(
  AntdUpload,
  mapReadPretty(PreviewText.Upload),
  mapProps(mapFormat())
)
渲染一个Upload组件 断点调试时，这块代码被重复调用了3 4次（只有一个 用到了上传组件的字段）

**Reply by XQ0118** [reply]
我也遇到了 connect 多次调用的问题

**Comment by XQ0118** [comment]
index.js:27 [Vue warn]: Invalid prop: type check failed for prop "onChange". Expected Function, got Array

这个问题：我的解决方案是在组件外面再套一层 div
